Validation/RecoTau/src/TauTagMCValidation.cc

```diff
@@ -0,0 +1,597 @@
+        nIsolated_NoChargedNoGammas_GammasSignal_                =ibooker.book1D(DiscriminatorLabel + "_GammasSignal",DiscriminatorLabel + "_GammasSignal",21, -0.5, 20.5);
+        nIsolated_NoChargedNoGammas_NeutralHadronsSignal_	       =ibooker.book1D(DiscriminatorLabel + "_NeutralHadronsSignal",DiscriminatorLabel + "_NeutralHadronsSignal",21, -0.5, 20.5);
+        nIsolated_NoChargedNoGammas_NeutralHadronsIsolAnnulus_   =ibooker.book1D(DiscriminatorLabel + "_NeutralHadronsIsolAnnulus",DiscriminatorLabel + "_NeutralHadronsIsolAnnulus",21, -0.5, 20.5);
+      }
+    }
+  }
+}
+
+void TauTagMCValidation::dqmBeginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) {
+  if (genericTriggerEventFlag_) {
+    if (genericTriggerEventFlag_->on()) {
+      genericTriggerEventFlag_->initRun(iRun, iSetup);
+    }
+  }
+}
+
+void TauTagMCValidation::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  if (genericTriggerEventFlag_) {
+    if (!genericTriggerEventFlag_->on()) std::cout<<"TauTagMCValidation::analyze: No working genericTriggerEventFlag. Did you specify a valid globaltag?"<<std::endl;//move to LogDebug?
+  }
+
+  numEvents_++;
+  double matching_criteria = -1.0;
+
+  //Initialize the Tau Multiplicity Counter
+  for ( std::vector< edm::ParameterSet >::iterator it = discriminators_.begin(); it!= discriminators_.end();  it++)
+  {
+    string DiscriminatorLabel = it->getParameter<string>("discriminator");
+    tauDeacyCountMap_.insert(std::make_pair("allHadronic" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("oneProng0Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("oneProng1Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("oneProng2Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("twoProng0Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("twoProng1Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("twoProng2Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("threeProng0Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.insert(std::make_pair("threeProng1Pi0" + DiscriminatorLabel, 0.));
+  }
+
+  typedef edm::View<reco::Candidate> genCandidateCollection;
+
+  // ----------------------- Reference product -----------------------------------------------------------------------
+
+  Handle<genCandidateCollection> ReferenceCollection;
+  bool isGen = iEvent.getByToken( refCollectionInputTagToken_, ReferenceCollection );
+
+  Handle<VertexCollection> pvHandle;
+  iEvent.getByToken( primaryVertexCollectionToken_, pvHandle ); //TO-DO
+
+  if (!isGen) {
+    std::cerr << " Reference collection: " << refCollection_ << " not found while running TauTagMCValidation.cc " << std::endl;
```

Comment:
Please replace std:cerr with LogWarning. Rule of thumb: never use std:cout/cerr in CMSSW.